cmd/talosctl/cmd/mgmt/kube_inject.go

```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package mgmt
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/talos-systems/talos/pkg/kubernetes/inject"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+var kubeInjectCmdFlags struct {
+	file  string
+	roles []string
+}
+
+var kubeInjectCmd = &cobra.Command{
+	Use:     "kube-inject -f <manifest.yaml> [--roles='<ROLE_1>,<ROLE_2>']",
+	Short:   "Inject talos API access into Kubernetes manifests",
+	Long:    `kube-inject reads a Kubernetes manifest and injects talos API access resources into it.`,
+	Example: `kube-inject -f deployment.yaml --roles="os:admin" > deployment-injected.yaml`,
+	Args:    cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+
+		reader := os.Stdin
+
+		if kubeInjectCmdFlags.file != "-" {
+			reader, err = os.Open(kubeInjectCmdFlags.file)
+			if err != nil {
+				return err
+			}
+		}
+
+		injectedYaml, err := inject.ServiceAccount(reader, kubeInjectCmdFlags.roles)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(string(injectedYaml))
+
+		return nil
+	},
+}
+
+func init() {
+	kubeInjectCmd.Flags().StringVarP(&kubeInjectCmdFlags.file, "file", "f", "-",
+		fmt.Sprintf("file with Kubernetes manifests to be injected with %s", constants.ServiceAccountResourceKind))
+	kubeInjectCmd.Flags().StringSliceVarP(&kubeInjectCmdFlags.roles, "roles", "r", []string{"os:reader"},
+		fmt.Sprintf("roles to add to the generated %s manifests", constants.ServiceAccountResourceKind))
+	addCommand(kubeInjectCmd)
```

Comment:
it might make sense to put this under `talosconfig`, i.e. `talosctl talosconfig kube-inject`

Reply:
Hm, I think it can be too long that way. Also, `talosconfig` command would be kind of redundant. But if you are sure, it's fine by me.

Reply:
Corrected it and hopefully made it more clear.

Reply:
my reason is the following: `talosctl kube-inject` is ambigious, as it's not clear what it injects, and sooner or later we might want another way of `kube-inject` (to inject something else into kubernetes pods). `talosctl talosconfig` is somewhat specific to the `talosconfig` (client config), so it falls under proper domain.

any other opinions?

Reply:
Yes, this makes sense. Another idea would be to switch it to `inject` and make `talosconfig` a subcommand or so (or rather, `serviceaccount`, since it is the type that is being injected. Sth like `talosctl inject serviceaccount -f ...`). 

This way, later, we'd be able to introduce `talosctl inject somethingelse`...

Reply:
yep, `talosctl inject serviceaccount` sounds good to me!

Reply:
done the change, feel free to have another look!